web/profiles/custom/os2loop/modules/os2loop_media/src/Helper/Helper.php

```diff
@@ -32,15 +33,112 @@ public function __construct(AccountProxyInterface $currentUser) {
    *
    * Change query for media library.
+   * The view config sets condition for viewing only media created by users with
+   * more than the authenticated role.
+   *
+   * @todo Move the role condition into this alter query and add a specific permission for viewing media added by users with multple roles.
+   *
+   * @param \Drupal\views\ViewExecutable $view
+   *   The view.
+   * @param \Drupal\views\Plugin\views\query\QueryPluginBase $query
+   *   The query.
    */
   public function queryAlter(ViewExecutable $view, QueryPluginBase $query) {
-    if ($query instanceof Sql && 'media_library' === $view->id()) {
-      // Add condition if user does not have permission to view all files.
-      if (!$this->currentUser->hasPermission('view all files in media browser')) {
+    if ($query instanceof Sql && ('media_library' === $view->id()) || ('media' === $view->id())) {
+      $currentUserRoles = $this->currentUser->getRoles();
+
+      // Allow display of all files for users with permission.
+      if ($this->currentUser->hasPermission('view all files in media browser')) {
+        $query = $this->removeRoleRestriction($query);
+      }
+
+      // Show only own files for users that are just authenticated
+      // (but not user 1).
+      if (1 === count($currentUserRoles) && in_array('authenticated', $currentUserRoles) && '1' !== $this->currentUser->id()) {
         $groupId = $query->setWhereGroup();
-        $query->addWhere($groupId, 'uid', $this->currentUser->id(), '=');
+        // @phpstan-ignore-next-line
+        $query->addWhere($groupId, 'media_field_data.uid', $this->currentUser->id(), '=');
+        $query = $this->removeRoleRestriction($query);
       }
     }
   }
 
+  /**
+   * Remove role restriction from query.
+   *
+   * @param \Drupal\views\Plugin\views\query\QueryPluginBase $query
+   *   The query.
+   *
+   * @property mixed $where
+   *
+   * @return \Drupal\views\Plugin\views\query\QueryPluginBase
+   *   The modified query.
+   */
+  private function removeRoleRestriction(QueryPluginBase $query): QueryPluginBase {
+    // Remove role restriction.
+    // @phpstan-ignore-next-line
+    foreach ($query->where[1]['conditions'] as $key => $condition) {
+      if ('users_field_data_media_field_data__user__roles.roles_target_id' === $condition['field']) {
+        // @phpstan-ignore-next-line
+        unset($query->where[1]['conditions'][$key]);
+      }
+    }
+
+    return $query;
+  }
+
+  /**
+   * Implements hook_form_alter().
+   *
+   * Alter media add/edit forms.
+   */
+  public function alterForm(array &$form, FormStateInterface $form_state, string $form_id) {
+    switch ($form_id) {
+      case 'media_os2loop_media_file_add_form':
+      case 'media_os2loop_media_image_add_form':
+      case 'media_os2loop_media_file_edit_form':
+      case 'media_os2loop_media_image_edit_form':
+      case 'media_library_add_form_upload':
+        $form['#current_user'] = $this->currentUser;
```

Comment:
Do we need to add the current user to the form? Can't we just use`$this->currentUser` for the checks and maybe add a `userCanSeeAllMedia` convenience function in the `Helper` class rather than duplicate the code?